src/insertWord.js

```diff
@@ -17,3 +18,28 @@ insertUi.addWordBtn.addEventListener("click", () => {
 
   insertUi.clear();
 });
+
+insertUi.doneBtn.addEventListener('click', (e) => {
+  if (addWordSuccess) {
+    const word = insertUi.createWordForUser(JSON.parse(localStorage.getItem(LOCALSTORAGE_KEY)));
+    wordService.saveWord(word);
+  
+    insertUi.clear();
+    addWordSuccess = null;
+  } else if (addWordSuccess === null) {
+    e.preventDefault();
+    const inputPass = insertUi.checkInputs();
+    if (!inputPass) {
+      insertUi.showAlert("Please fill the word input and make a recording or fill in the definition or example sentence input", "danger");
+      addWordSuccess = false;
+      insertUi.doneBtn.click();
+      return;
```

Comment:
if `inputPass` fails, it is not necessary to click the `doneBtn` again. You only need prevent the default and show the alert.